assets/queries/cloudFormation/s3_buckets_are_publicly_accessible/query.rego

```diff
@@ -0,0 +1,50 @@
+package Cx
+
+CxPolicy [ result ] {
+ 	resourcePolicy := input.document[indexPolicy].Resources[namePolicy]
+	policyProperties := [policyProperties | resourcePolicy.Type == "AWS::IAM::Policy"; policyProperties = resourcePolicy.Properties.PolicyDocument]
+	policyStatements := policyProperties[_].Statement[index]
+    
+  resourceBucket := input.document[indexBucket].Resources[nameBucket]
+  bucketProperties := [bucketProperties | resourceBucket.Type == "AWS::S3::Bucket"; bucketProperties = resourceBucket.Properties]
+
+ 	checkPolicyConfiguration(policyStatements)
+  checkPublicAccessBlockConfiguration(bucketProperties)
+  
+	result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("Resources.%s.Properties.PublicAccessBlockConfiguration", [nameBucket]),
+                "issueType":		"IncorrectValue",  
+                "keyExpectedValue": "'Resources.Properties.PublicAccessBlockConfiguration' is setted and configuration has value true",
+                "keyActualValue": 	"'Resources.Properties.PublicAccessBlockConfiguration' is not setted or configuration has value false "
+                }
+}
+
+checkPublicAccessBlockConfiguration(bucketProperties) = true {
+	checkAccess(bucketProperties[_].PublicAccessBlockConfiguration)
+}
+checkAccess(property){
+	property != null
+	property.RestrictPublicBuckets != true
+}
+checkAccess(property){
+	property != null
+    property.IgnorePublicAcls != true
+}
+
+checkPolicyConfiguration(policyProperties) {
+    policyProperties.Effect == "Allow"
+    checkPolicy(policyProperties)
+}
```

Comment:
This fuction will receive an array of Policy Statements.
Replace with:
```
checkPolicyConfiguration(policyStatements) {
    some p
    policyStatements[p].Effect == "Allow"
    checkPolicy(policyStatements[p])
}
```
It will return true if any of the statements has [Effect='Allow' and (Principal='\*' or Principal.AWS='\*')]

Reply:
Done